src/rules/non-interactive-tooltip-trigger.js

```diff
@@ -0,0 +1,126 @@
+const {isPrimerComponent} = require('../utils/is-primer-component')
+const {getJSXOpeningElementName} = require('../utils/get-jsx-opening-element-name')
+const {getJSXOpeningElementAttribute} = require('../utils/get-jsx-opening-element-attribute')
+
+isInteractive = child => {
+  const childName = getJSXOpeningElementName(child.openingElement)
+  return ['button', 'summary', 'select', 'textarea', 'a', 'input', 'iconbutton'].includes(childName.toLowerCase())
+}
+
+isInteractiveAnchor = child => {
+  const hasHref = getJSXOpeningElementAttribute(child.openingElement, 'href')
+  if (!hasHref) return false
+  const href = getJSXOpeningElementAttribute(child.openingElement, 'href').value.value
+  const isAnchorInteractive = typeof href === 'string' && href !== ''
+  return isAnchorInteractive
+}
+
+isInteractiveInput = child => {
+  const hasHiddenType =
+    getJSXOpeningElementAttribute(child.openingElement, 'type') &&
+    getJSXOpeningElementAttribute(child.openingElement, 'type').value.value === 'hidden'
+  return !hasHiddenType
+}
+
+const checkTriggerElement = jsxNode => {
+  let messageId = ''
+  const child = jsxNode.children
+  const childName = getJSXOpeningElementName(child.openingElement)
+
+  // First check specific requirements for anchor
+  if (childName === 'a' && !isInteractiveAnchor(child)) {
+    messageId = 'anchorTagWithoutHref'
+    return {messageId, node: jsxNode}
+  }
+  // Then check specific requirements input
+  if (childName === 'input' && !isInteractiveInput(child)) {
+    messageId = 'hiddenInput'
+    return {messageId, node: jsxNode}
+  }
+  // Then check if the child is interactive
+  if (!isInteractive(child)) {
+    // If child is not interactive, check if there are any grandchildren that is interactive
+    const hasJsxGrands =
+      child.children.length > 0 && child.children.filter(gChild => gChild.type === 'JSXElement').length > 0 //.some(gChild => isInteractive(gChild))
+
+    if (!hasJsxGrands) {
+      messageId = 'nonInteractiveTrigger'
+    } else {
+      const hasInteractiveGrands = child.children // is there any way I can access all child nodes? :/
+        .filter(gChild => gChild.type === 'JSXElement')
+        .some(gChild => {
+          const gChildName = getJSXOpeningElementName(gChild.openingElement)
+          // TODO: How can I check all child nodes?
```

Comment:
I am trying to understand if I should check grandchildrens' interactivity? 🤔 And is so, is there a way to access all descendants here?

Reply:
I definitely think it can be helpful to recursively check children but unfortunately we're limited to the owner tree and won't be able to see the render result of children 😞 Definitely worth the trade-off though, we just may error out incorrectly if a component renders something interactive.

One quick way to get a flat list of all the jsxElement children could be a pre-order traversal:

```jsx
function getChildren(node) {
  if (Array.isArray(node.children)) {
    return node.children.filter((child) => {
      return child.type === 'JSXElement';
    }).flatMap((child) => {
      return [child, ...getChildren(child)];
    });
  }
  return [];
}
```

Hope this works 😅

Reply:
>I definitely think it can be helpful to recursively check children but unfortunately we're limited to the owner tree and won't be able to see the render result of children 😞

Yeah, it makes things harder but that is okay. The main reason I tried recursive, I wanted to get all children as a flat list and that function you wrote works perfectly for this ✨